### ISSUE TITLE
tmux 3.1c - unable to connect to process

### ISSUE THREAD

**Issue by tchan09** [SynoCommunity/spksrc#4467]
### Setup
_Package Name:_ `SynoCli Network Tools`
_Package Version:_ `1.5-8`

_NAS Model:_ `DS918+`
_NAS Architecture:_ `apollolake`
_DSM version:_ `6.2.3-25426 Update 2`

### Expected behavior
After updating to tmux 3.1c via the new SynoCli Network Tools package, I noticed tmux was acting funny. The `tpm` plugin manager failed to start and digging deeper, it seems to be related to permission settings that prevent connecting to the running tmux daemon. I have tried killing all tmux processes, uninstalling the SynoCli Network Tools package and reinstalling the package.

For example, expected behaviour:
```bash
# From within a running tmux session
> tmux ls
0: 5 windows (created Sun Jan 10 15:59:31 2021) (attached)
```

### Actual behavior
```bash
> tmux ls
error connecting to /volume3/@appstore/synocli-net/orig/tmux (Permission denied)
```

### Steps to reproduce
_1._ Start tmux
_2._ Try to run any tmux command, such as listing sessions via `tmux ls`

#### Package log
```
 Tue Mar  2 13:17:02 EST 2021
 ===> Step preinst. USER= GROUP= SHARE_PATH=
 Tue Mar  2 13:17:03 EST 2021
 ===> Step postinst. USER= GROUP= SHARE_PATH=
 Invoke service_create_links
 create link: /usr/local/bin/screen -> /volume3/@appstore/synocli-net/bin/screen
 create link: /usr/local/bin/tmux -> /volume3/@appstore/synocli-net/bin/tmux
 create link: /usr/local/bin/nmap -> /volume3/@appstore/synocli-net/bin/nmap
 create link: /usr/local/bin/nping -> /volume3/@appstore/synocli-net/bin/nping
 create link: /usr/local/bin/ndiff -> /volume3/@appstore/synocli-net/bin/ndiff
 create link: /usr/local/bin/arp-scan -> /volume3/@appstore/synocli-net/bin/arp-scan
 create link: /usr/local/bin/arp-fingerprint -> /volume3/@appstore/synocli-net/bin/arp-fingerprint
 create link: /usr/local/bin/get-iab -> /volume3/@appstore/synocli-net/bin/get-iab
 create link: /usr/local/bin/get-oui -> /volume3/@appstore/synocli-net/bin/get-oui
 create link: /usr/local/bin/links -> /volume3/@appstore/synocli-net/bin/links
 create link: /usr/local/bin/sshfs -> /volume3/@appstore/synocli-net/bin/sshfs
 create link: /usr/local/bin/fusermount -> /volume3/@appstore/synocli-net/bin/fusermount
 create link: /usr/local/bin/socat -> /volume3/@appstore/synocli-net/bin/socat
 create link: /usr/local/bin/procan -> /volume3/@appstore/synocli-net/bin/procan
 create link: /usr/local/bin/filan -> /volume3/@appstore/synocli-net/bin/filan
 create link: /usr/local/bin/fritzctl -> /volume3/@appstore/synocli-net/bin/fritzctl
 create link: /usr/local/bin/rsync -> /volume3/@appstore/synocli-net/bin/rsync
 create link: /usr/local/bin/autossh -> /volume3/@appstore/synocli-net/bin/autossh
 create link: /usr/local/bin/scp -> /volume3/@appstore/synocli-net/bin/scp
 create link: /usr/local/bin/sftp -> /volume3/@appstore/synocli-net/bin/sftp
 create link: /usr/local/bin/slogin -> /volume3/@appstore/synocli-net/bin/slogin
 create link: /usr/local/bin/ssh -> /volume3/@appstore/synocli-net/bin/ssh
 create link: /usr/local/bin/ssh-add -> /volume3/@appstore/synocli-net/bin/ssh-add
 create link: /usr/local/bin/ssh-agent -> /volume3/@appstore/synocli-net/bin/ssh-agent
 create link: /usr/local/bin/ssh-keygen -> /volume3/@appstore/synocli-net/bin/ssh-keygen
 create link: /usr/local/bin/ssh-keyscan -> /volume3/@appstore/synocli-net/bin/ssh-keyscan
 create link: /usr/local/libexec/sftp-server -> /volume3/@appstore/synocli-net/libexec/sftp-server
 create link: /usr/local/libexec/ssh-keysign -> /volume3/@appstore/synocli-net/libexec/ssh-keysign
 create link: /usr/local/libexec/ssh-pkcs11-helper -> /volume3/@appstore/synocli-net/libexec/ssh-pkcs11-helper
 create link: /usr/local/libexec/ssh-sk-helper -> /volume3/@appstore/synocli-net/libexec/ssh-sk-helper
 create link: /usr/local/sbin/sshd -> /volume3/@appstore/synocli-net/sbin/sshd
 create link: /usr/local/sbin/ser2net -> /volume3/@appstore/synocli-net/sbin/ser2net
```
#### Other logs
_`/var/log/synopkg.log`_
```
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin preinst
 2021/03/02 13:17:02 install synocli-net 1.5-8 End preinst ret=[0]
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin /bin/rm -rf /volume3/@appstore/synocli-net
 2021/03/02 13:17:02 install synocli-net 1.5-8 End /bin/rm -rf /volume3/@appstore/synocli-net ret=[0]
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin /bin/mv -f /volume3/@tmp/pkginstall/package /volume3/@appstore/synocli-net
 2021/03/02 13:17:02 install synocli-net 1.5-8 End /bin/mv -f /volume3/@tmp/pkginstall/package /volume3/@appstore/synocli-net ret=[0]
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin /bin/rm -rf /var/packages/synocli-net
 2021/03/02 13:17:02 install synocli-net 1.5-8 End /bin/rm -rf /var/packages/synocli-net ret=[0]
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin /bin/mkdir -p /var/packages/synocli-net
 2021/03/02 13:17:02 install synocli-net 1.5-8 End /bin/mkdir -p /var/packages/synocli-net ret=[0]
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin /bin/mv -f /volume3/@tmp/pkginstall/INFO /var/packages/synocli-net/INFO
 2021/03/02 13:17:02 install synocli-net 1.5-8 End /bin/mv -f /volume3/@tmp/pkginstall/INFO /var/packages/synocli-net/INFO ret=[0]
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin /bin/rm -rf /var/packages/synocli-net/scripts
 2021/03/02 13:17:02 install synocli-net 1.5-8 End /bin/rm -rf /var/packages/synocli-net/scripts ret=[0]
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin /bin/mv -f /volume3/@tmp/pkginstall/scripts /var/packages/synocli-net/scripts
 2021/03/02 13:17:02 install synocli-net 1.5-8 End /bin/mv -f /volume3/@tmp/pkginstall/scripts /var/packages/synocli-net/scripts ret=[0]
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin /bin/rm -rf /var/packages/synocli-net/conf
 2021/03/02 13:17:02 install synocli-net 1.5-8 End /bin/rm -rf /var/packages/synocli-net/conf ret=[0]
 2021/03/02 13:17:02 install synocli-net 1.5-8 Begin /bin/mv -f /volume3/@tmp/pkginstall/conf /var/packages/synocli-net/conf
 2021/03/02 13:17:02 install synocli-net 1.5-8 End /bin/mv -f /volume3/@tmp/pkginstall/conf /var/packages/synocli-net/conf ret=[0]
 2021/03/02 13:17:03 install synocli-net 1.5-8 Begin postinst
 2021/03/02 13:17:03 install synocli-net 1.5-8 End postinst ret=[0]
 2021/03/02 13:17:05 install synocli-net: begin to start version 1.5-8
 2021/03/02 13:17:05 install synocli-net 1.5-8 Begin pre-load apparmor
 2021/03/02 13:17:05 install synocli-net 1.5-8 End pre-load apparmor ret=[0]
 2021/03/02 13:17:05 install synocli-net 1.5-8 Begin start-stop-status start
 2021/03/02 13:17:05 install synocli-net 1.5-8 End start-stop-status start ret=[0]
 2021/03/02 13:17:05 install synocli-net: start version 1.5-8 successfully, result 0
 2021/03/02 13:17:05 install synocli-net 1.5-8 Begin /bin/rm -rf /volume3/@tmp/pkginstall
 2021/03/02 13:17:05 install synocli-net 1.5-8 End /bin/rm -rf /volume3/@tmp/pkginstall ret=[0]
 2021/03/02 13:17:05 install synocli-net 1.5-8 successfully
```

**Comment by etcusrvar** [comment]
It may be an arch specific issue. Doing the same steps on avoton works fine for me. Did tmux 2.9a in synocli-net 1.5-7 have this issue? You could downgrade manually by getting the old package from https://synocommunity.com/packages. I authored #4438 which did the upgrade. At least the first half of it. @hgy59, I'm unclear what the other commit in that PR was for.

**Comment by tchan09** [comment]
I can confirm that the issue goes away if I downgrade to 1.5-7 and run with tmux 2.9a. I'm also running tmux 3.1c on other non-Synology servers with the same conf file and haven't had issues elsewhere.

Interesting the package works on avoton. At first I thought it might be the change with the utf-8 wrapper script and the placement of tmux in `orig` instead of `bin` but that wouldn't explain why it works on avoton.

At this point, I'm going to stick with 1.5-7 and monitor this thread. I have both spks saved locally and am happy to switch between them for debugging.

**Comment by hgy59** [comment]
> Interesting the package works on avoton. At first I thought it might be the change with the utf-8 wrapper script and the placement of tmux in `orig` instead of `bin` but that wouldn't explain why it works on avoton.

The utc-8 wrapper is my suspicion too. To test this, you could copy tmux from orig to bin folder (if you want to keep the utf-8 wrapper bin/tmux you have to rename it before).
An maybe it is worth to test running tmux as /var/packages/synocli-net/target/bin/tmux instead of using the link in /usr/local/bin/tmux.

**Comment by tchan09** [comment]
> The utc-8 wrapper is my suspicion too. To test this, you could copy tmux from orig to bin folder (if you want to keep the utf-8 wrapper bin/tmux you have to rename it before).

Running the tmux binary from `bin` instead of using the shell script to run it from `orig` did the trick, thanks! I have no idea why though.

> An maybe it is worth to test running tmux as /var/packages/synocli-net/target/bin/tmux instead of using the link in /usr/local/bin/tmux.

I also tried this but this didn't work for me.

**Comment by phinze** [comment]
Hi folks! Identical setup w/ identical behavior as @tchan09 has reported. Unwinding the utf-8 wrapper also worked around the issue for me. Here are the commands I ran:

```bash
# Copy tmux binary from orig to bin, preserving script as backup w/ .utf8-orig extension
sudo cp -v /volume1/@appstore/synocli-net/bin/tmux{,.utf8-orig}
sudo cp -v /volume1/@appstore/synocli-net/{orig,bin}/tmux
```

Having dug into the error message a little, it seems like it's looking in the wrong place for the tmux socket. The error message suggests it's treating the script path as the socket and trying to connect to it.

If you manually specify the socket via `tmux -S` it also works around the issue:

```sh
# Works fine
tmux -S $(echo $TMUX | cut -d, -f1) ls
```

**Comment by hgy59** [comment]
As the utf-8 wrapper seems to be outdated and obsolete, I will remove it.

As all my available Diskstations with DSM 5.2/6.1/7.0 have the `LC_ALL="en_US.utf8"` exported, this kind of wrapper sould not be required anymore. 
If for any reason the utf-8 characterset is not the default, one have to call `export LC_ALL="en_US.utf8"` before calling tmux.
